src/MagnumPlugins/AssimpImporter/AssimpImporter.cpp

```diff
@@ -0,0 +1,526 @@
+
+    if(mat->Get(AI_MATKEY_TEXTURE(aiTextureType_AMBIENT, 0), texturePath) == AI_SUCCESS) {
+        flags |= PhongMaterialData::Flag::AmbientTexture;
+    }
+    if(mat->Get(AI_MATKEY_TEXTURE(aiTextureType_DIFFUSE, 0), texturePath) == AI_SUCCESS) {
+        flags |= PhongMaterialData::Flag::DiffuseTexture;
+    }
+    if(mat->Get(AI_MATKEY_TEXTURE(aiTextureType_SPECULAR, 0), texturePath) == AI_SUCCESS) {
+        flags |= PhongMaterialData::Flag::SpecularTexture;
+    }
+    /* @todo many more types supported in assimp */
+
+    mat->Get(AI_MATKEY_SHININESS, shininess); /* Key always present, default 0.0f */
+
+    std::unique_ptr<PhongMaterialData> data{new PhongMaterialData(flags, shininess, mat)};
+
+    mat->Get(AI_MATKEY_COLOR_AMBIENT, color); /* Key always present, default black */
+    if(flags & PhongMaterialData::Flag::AmbientTexture)
+        data->ambientColor() = Color3(color);
+
+    mat->Get(AI_MATKEY_COLOR_DIFFUSE, color); /* Key always present, default black */
+    if(flags & PhongMaterialData::Flag::DiffuseTexture)
+        data->diffuseColor() = Color3(color);
+
+    mat->Get(AI_MATKEY_COLOR_SPECULAR, color); /* Key always present, default black */
+    if(flags & PhongMaterialData::Flag::SpecularTexture)
+        data->specularColor() = Color3(color);
+
+    return std::move(data);
+}
+
+UnsignedInt AssimpImporter::doTextureCount() const { return _textures.size(); }
+
+std::optional<TextureData> AssimpImporter::doTexture(const UnsignedInt id) {
+    constexpr Sampler::Filter filterU = Sampler::Filter::Linear;
+    constexpr Sampler::Filter filterV = Sampler::Filter::Linear;
+
+    /* Load texture wrapping property */
+    Sampler::Wrapping wrappingU = Sampler::Wrapping::ClampToEdge;
+    Sampler::Wrapping wrappingV = Sampler::Wrapping::ClampToEdge;
+    constexpr Sampler::Wrapping wrappingW = Sampler::Wrapping::ClampToEdge;
+
+    auto toWrapping = [](aiTextureMapMode mapMode) {
+        switch (mapMode) {
+        case aiTextureMapMode_Wrap:
+            return Sampler::Wrapping::Repeat;
+        case aiTextureMapMode_Decal:
+            Warning() << "Trade::AssimpImporter::texture(): no wrapping enum to match aiTextureMapMode_Decal, using Sampler::Wrapping::ClampToEdge.";
+        case aiTextureMapMode_Clamp:
+            return Sampler::Wrapping::ClampToEdge;
+        case aiTextureMapMode_Mirror:
+            return Sampler::Wrapping::MirroredRepeat;
+        }
+    };
+
+    aiTextureMapMode mapMode;
+    const aiMaterial* mat = _textures[id].first;
+    const aiTextureType type = _textures[id].second;
+    if(mat->Get(AI_MATKEY_MAPPINGMODE_U(type, 0), mapMode) == AI_SUCCESS) {
+        wrappingU = toWrapping(mapMode);
+    }
+    if(mat->Get(AI_MATKEY_MAPPINGMODE_V(type, 0), mapMode) == AI_SUCCESS) {
+        wrappingV = toWrapping(mapMode);
+    }
+
+    return TextureData{TextureData::Type::Texture2D, filterU, filterV, {Sampler::Mipmap::Linear}, {wrappingU, wrappingV, wrappingW}, id, &_textures[id]};
```

Comment:
I am expecting this may be loadable some day, or I possibly missed something. Then it's easier to add...
Since the constants are rather long, it makes that return statement rather long and less readible. As the compiler will optimize it away, I didn't see any harm :D